netbox_diode_plugin/api/applier.py

```diff
@@ -45,42 +45,59 @@ def apply_changeset(change_set: ChangeSet) -> ChangeSetResult:
         id=change_set.id,
     )
 
-def _apply_change(data: dict, model_class: models.Model, change: Change, created: dict):
+def _apply_change(data: dict, model_class: models.Model, change: Change, created: dict, request):
     serializer_class = get_serializer_for_model(model_class)
     change_type = change.change_type
+    logger.error(f"applying {data}")
     if change_type == ChangeType.CREATE.value:
-        serializer = serializer_class(data=data)
+        serializer = serializer_class(data=data, context={"request": request})
```

Comment:
some referential field types require the request to be available to the serializer (hyperlinked-something-or-other used by custom field object references)